lib/json_schemer/schema/base.rb

```diff
@@ -598,7 +602,7 @@ def safe_strict_decode64(data)
 
       def ecma_262_regex(pattern)
         @ecma_262_regex ||= {}
-        @ecma_262_regex[pattern] ||= Regexp.new(
+        @ecma_262_regex[pattern] ||= regexp_resolver.call(
```

Comment:
I think we could reuse `CachedRefResolver` to handle this caching. That way the resolver would be responsible for caching and it could be customized/removed if people wanted to. Maybe rename it to `CachedResolver` and turn `Regexp::Scanner` portion into a built-in `regexp_resolver`.